survey/survey_app/serializers.py

```diff
@@ -30,19 +33,52 @@ class Meta:
 
 
 class AnswerSerializer(serializers.ModelSerializer):
+    # Add the 'required=True' parameter to the 'question' field
+    question = serializers.PrimaryKeyRelatedField(queryset=Question.objects.all(), required=True)
+    option = serializers.PrimaryKeyRelatedField(queryset=Option.objects.all())
+
     class Meta:
         model = Answer
-        fields = '__all__'
+        fields = ['question', 'option']
+
+    def create(self, validated_data):
+        user = self.context['request'].user
+        if user:
+            # Get the current date
+            today = timezone.now().date()
+
+            # Count the number of answers submitted by the user on the current date
+            user_submission_count = Answer.objects.user_submission_count_for_day(user, today)
```

Comment:
We can just use a util function rather than defining a custom model manager @newDaria

Reply:
Yes we can update def in models.py
 ```
   @staticmethod
    def user_submission_count_for_day(user, date):
        return Answer.objects.filter(creator=user, created_at__date=date).count()

then in serializers.py 
            user_submission_count = Answer.user_submission_count_for_day(user, today)
```